src/components/engagement/engagement.service.ts

```diff
@@ -358,12 +358,17 @@ export class EngagementService {
     );
 
     try {
-      if (mentorId) {
-        await this.repo.updateMentor(input.id, mentorId);
+      if (mentorId !== undefined) {
+        await this.repo.updateRelation('mentor', 'User', input.id, mentorId);
       }
 
-      if (countryOfOriginId) {
-        await this.repo.updateCountryOfOrigin(input.id, countryOfOriginId);
+      if (countryOfOriginId !== undefined) {
+        await this.repo.updateRelation(
+          'countryOfOrigin',
+          'User',
```

Comment:
```suggestion
          'Location',
```
I know, the inputs are confusing. This is going away with EdgeDB.

Reply:
After reviewing the older PR that your pointed me at I see what you mean